BondageClub/Scripts/Dialog.js

```diff
@@ -1002,28 +1011,65 @@ function DialogFacialExpressionsLoad(Slot) {
 		DialogFacialExpressionsBuild();
 	}
 }
-/**draws the savedexpressions menu */
+
+/**
+ * Builds the savedexpressions menu previews.
+ * @returns {void} - Nothing
+ */
+function DialogBuildSavedExpressionsMenu() {
+	const ExcludedGroups = ["Mask"];
+	const BaseAppearance = Player.Appearance.filter(A => A.Asset.Group.Category === "Appearance" && !ExcludedGroups.includes(A.Asset.Group.Name) &&  !A.Asset.Group.AllowExpression);
+	Player.SavedExpressions.forEach((expression, i) => {
+		if (expression) {
+			const PreviewCharacter = CharacterLoadSimple("SavedExpressionPreview-" + i);
+			PreviewCharacter.Appearance = BaseAppearance.slice();
+
+			for (let x = 0; x < expression.length; x++) {
+				CharacterSetFacialExpression(PreviewCharacter, expression[x].Group, expression[x].CurrentExpression);
+			}
+
+			DialogSavedExpressionPreviews[i] = PreviewCharacter;
+		}
+	});
+}
+
+/**
+ * Draws the savedexpressions menu
+ * @returns {void} - Nothing
+ */
 function DialogDrawSavedExpressionsMenu() {
-	DrawText(DialogFindPlayer("SavedExpressions"), 195, 25, "White", "Black");
-	DrawText(DialogFindPlayer("SavedExpressionsSave"), 140, 180, "White", "Black");
-	DrawText(DialogFindPlayer("SavedExpressionsLoad"), 260, 180, "White", "Black");
+	DrawText(DialogFindPlayer("SavedExpressions"), 210, 90, "White", "Black");
+
+	if (!DialogSavedExpressionPreviews || !DialogSavedExpressionPreviews.length || Player.SavedExpressions.some(expression => expression != null))
```

Comment:
The menu build is now occurring every draw, in _both_ the saved expressions being empty and non-empty cases. I was thinking something more along the lines of:
```suggestion
	if (!DialogSavedExpressionPreviews || (!DialogSavedExpressionPreviews.length && Player.SavedExpressions.length))
```